src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,997 @@
+                                measure_pulse_amplitude=None,
+                                acquire_delay=self.time_of_flight,
+                                reset_delay=relaxation_time,
+                            )
+                            i += 1
+
+    def fast_reset(self, exp, qubits, fast_reset):
+        """fast reset after readout - small delay for signal processing"""
+        print("Im fast resetting")
+        print(fast_reset)
+        print(self.sequence_qibo.qubits)
+        for qubit_name in self.sequence_qibo.qubits:
+            qubit = qubits[qubit_name]
+            if not qubit.flux_coupler:
+                with exp.section(uid=f"fast_reset{qubit.name}", play_after=f"sequence_measure{qubit.name}"):
+                    with exp.match_local(handle=f"acquire{qubit.name}"):
+                        with exp.case(state=0):
+                            pass
+                        with exp.case(state=1):
+                            exp.play(signal=f"drive{qubit.name}", pulse=ZhPulse(fast_reset[qubit.name]).zhpulse)
+
+    def sweep(self, qubits, sequence, *sweepers, options):
+        """Play pulse and sweepers sequence"""
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+
+        self.sweepers = list(sweepers)
+        self.sequence_zh(sequence, qubits, sweepers)
+        self.calibration_step(qubits)
+        self.create_exp(qubits, options)
+        self.run_exp()
+
+        # TODO: Test for several readouts and qubits simultaneosuly
+        results = {}
+        for qubit in qubits.values():
+            if not qubit.flux_coupler:
+                if self.sequence[f"readout{qubit.name}"]:
+                    exp_res = self.results.get_data(f"sequence{qubit.name}")
+                    i = exp_res.real
+                    q = exp_res.imag
+                    results[self.sequence[f"readout{qubit.name}"][0].pulse.serial] = ExecutionResults.from_components(
+                        i, q
+                    )
+
+        # self.run_sim(sim_time) #FIXME: Careful placement or reconnection to avoid messing with several executions
+        # lo.show_pulse_sheet("pulses", self.exp) #FIXME: Wait until next LO release for bugs
+        # self.disconnect() #FIXME: Careful placement or reconnection to avoid messing with several executions
+        return results
+
+    # TODO: Recursion tests and Better sweeps logic
+    def sweep_recursion(self, qubits, exp, exp_calib, relaxation_time, acquisition_type, fast_reset):
+        """Sweepers recursion for multiple nested sweepers"""
+        # Ordered like this for how they defined the frequncy sweep on qibocal to be the last
+        # and I need it to be the outer one
+        sweeper = self.sweepers[-1]
+        i = len(self.sweepers) - 1
+        self.sweepers.remove(sweeper)
+
+        parameter = None
+        print("sweep", sweeper.parameter.name.lower())
+
+        if sweeper.parameter.name.lower() == "frequency":
```

Comment:
```suggestion
        if sweeper.parameter is Parameter.frequency:
```
same for the other checks.